server/src/main/java/org/candlepin/async/JobManager.java

```diff
@@ -1539,6 +1539,14 @@ private AsyncJobStatus fetchJobStatus(JobMessage message)
             log.error(errmsg);
             throw new JobInitializationException(errmsg, true);
         }
+        else if (jobState == JobState.RUNNING) {
+            String errmsg =
+                String.format("Job \"%s\" (%s) is in the %s state, which is non-terminal and invalid",
```

Comment:
Can we change the error message to something that gives more information on the situation? Something along the lines of:

```"Job \"%s\" (%s) is already running on host \"%s\"; ignoring execution request"```

We could also add a test here where if the host equals the current host, we just manually set the job state to QUEUED and continue execution as normal.

Reply:
Done